pkg/controller/restapi/operation/vdri/operation.go

```diff
@@ -0,0 +1,97 @@
+/*
+ *
+ * Copyright SecureKey Technologies Inc. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * /
+ *
+ */
+
+package vdri
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/url"
+
+	"github.com/hyperledger/aries-framework-go/pkg/controller/command/vdri"
+	resterrors "github.com/hyperledger/aries-framework-go/pkg/controller/restapi/errors"
+	vdriapi "github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdri"
+	"github.com/hyperledger/aries-framework-go/pkg/internal/common/support"
+	"github.com/hyperledger/aries-framework-go/pkg/restapi/operation"
+)
+
+const (
+	vdriOperationID     = "/vdri"
+	createPublicDIDPath = vdriOperationID + "/create-public-did"
+)
+
+// provider contains dependencies for the common controller operations
+// and is typically created by using aries.Context()
+type provider interface {
+	VDRIRegistry() vdriapi.Registry
+}
+
+// Operation contains basic common operations provided by controller REST API
+type Operation struct {
+	handlers []operation.Handler
+	command  *vdri.Command
+}
+
+// New returns new common operations rest client instance
+func New(ctx provider) (*Operation, error) {
+	o := &Operation{command: vdri.New(ctx)}
+
+	defer o.registerHandler()
```

Comment:
Why defer?

Reply:
we don't need it anymore, will remove it.